feat(perf): Add project avatars to HTTP module

gggritso opened the pull request:
Hostnames are ambiguous, it can be important which project they belong to. Right now this is lost! Show project avatar next to domains where appropriate.

**e.g.,**
<img width="256" alt="Screenshot 2024-03-22 at 5 47 34 PM" src="https://github.com/getsentry/sentry/assets/989898/c3459df9-c4c0-4848-a385-d070e9aa8dda">
<img width="264" alt="Screenshot 2024-03-22 at 6 04 28 PM" src="https://github.com/getsentry/sentry/assets/989898/dd521ac9-2077-4b21-9542-b79bf6daa906">

- Use `useLocationQuery`
- Show project next to domain in table
- Show project next to hostname in summary

gggritso committed:
Show project next to domain in table

gggritso committed:
Use `useLocationQuery`

Safer than pulling directly off the location

gggritso committed:
Show project next to hostname in summary

AbhiPrasad commented on static/app/views/performance/http/httpDomainSummaryPage.tsx:
```diff
@@ -157,6 +169,7 @@ export function HTTPDomainSummaryPage() {
             ]}
           />
           <Layout.Title>
+            {project && <ProjectAvatar project={project} size={36} />}
```
<img width="338" alt="image" src="https://github.com/getsentry/sentry/assets/18689448/c2d634f5-adcb-4d14-83db-f23f4e920ebb">

I think this needs `align-items: center`, the avatar is mis-aligned with the name

gggritso committed:
Merge branch 'master' into feat/perf/http-module-domain-avatars

gggritso committed:
Vertically align avatar and hostname

gggritso committed:
Fix truncation

- make sure the domain name actually truncates with ellipsis
- prevent breaking onto new lines
